File: app/src/main/java/com/galib/natorepbs2/viewmodel/InformationViewModel.java
package com.galib.natorepbs2.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.galib.natorepbs2.constants.Category;
import com.galib.natorepbs2.db.Information;
import com.galib.natorepbs2.db.NPBS2Repository;

import java.util.ArrayList;
import java.util.List;

public class InformationViewModel extends AndroidViewModel {
    private NPBS2Repository mRepository;

    public InformationViewModel (Application application) {
        super(application);
        mRepository = new NPBS2Repository(application);
    }

    public LiveData<List<Information>> getInformationByCategory(String category) {
        return mRepository.getInformationByCategory(category);
    }

    public void setMonth(String month){
        mRepository.setMonth(month);
    }

    public LiveData<Information> getMonth(){
        return  mRepository.getMonth();
    }

    public void insertFromAtAGlance(List<List<String>> trtd) {
        List<Information> informationList = new ArrayList<>();
        for(int i =1; i<trtd.size(); i++){
            informationList.add(new Information(Integer.parseInt(trtd.get(i).get(0)), trtd.get(i).get(1), trtd.get(i).get(2), Category.atAGlance));
        }
        mRepository.insertInformations(informationList);
    }

    public void insertAll(List<Information> data){
        mRepository.insertInformations(data);
    }

    public void deleteAllByCategory(String atAGlance) {
        mRepository.deleteAllByCategory(Category.atAGlance);
    }

    public void insertVisionMission(String vision, String mission) {
        String s[] = vision.split(": ");
        mRepository.insertInformation(new Information(0, s[0], s[1], Category.visionMission));
        s = mission.split(": ");
        mRepository.insertInformation(new Information(1, s[0], s[1], Category.visionMission));
    }
}

File: app/src/main/java/com/galib/natorepbs2/viewmodel/InformationViewModel.kt
package com.galib.natorepbs2.viewmodel

import androidx.lifecycle.*
import com.galib.natorepbs2.db.NPBS2Repository
import com.galib.natorepbs2.models.Information
import com.galib.natorepbs2.models.Instruction
import kotlinx.coroutines.launch

class InformationViewModel(private val mRepository: NPBS2Repository) : ViewModel() {

    fun getInformationByCategory(category: String?): LiveData<List<Information>> {
        return mRepository.getInformationByCategory(category).asLiveData()
    }

    val month: LiveData<Information?>
        get() = mRepository.month.asLiveData()

    val importantNotice: LiveData<Information?>
        get() = mRepository.importantNotice.asLiveData()

    val bannerUrl: LiveData<List<String>>
        get() = mRepository.bannersUrl.asLiveData()

    fun insertAll(data: List<Information>) = viewModelScope.launch{
        mRepository.insertInformations(data)
    }

    fun getInstructionByType(type: Int): MutableList<Instruction>? {
        return mRepository.getInstructionByType(type)
    }
}

class InformationViewModelFactory(private val repository: NPBS2Repository) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(InformationViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return InformationViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}